packages/jest-core/src/ViteDevServerManager.ts

```diff
@@ -0,0 +1,414 @@
+
+      if (module) {
+        const collectDeps = (mod: any, visited = new Set<any>()): void => {
+          if (visited.has(mod)) {
+            return;
+          }
+          visited.add(mod);
+
+          for (const imported of mod.importedModules || []) {
+            if (imported.file) {
+              dependencies.add(imported.file);
+              collectDeps(imported, visited);
+            }
+          }
+        };
+
+        collectDeps(module);
+      }
+
+      this.moduleGraphCache.set(filePath, dependencies);
+    } catch (error: any) {
+      console.error(
+        `Failed to get dependencies for ${filePath}:`,
+        error.message,
+      );
+    }
+
+    return dependencies;
+  }
+
+  /**
+   * Clears all caches (transform cache and module graph cache)
+   */
+  clearCaches(): void {
+    this.transformCache.clear();
+    this.moduleGraphCache.clear();
+  }
+
+  /**
+   * Gets the Vite module graph for dependency tracking
+   */
+  getModuleGraph(): any {
+    return this.viteDevServer?.moduleGraph;
+  }
+
+  /**
+   * Checks if the dev server is running
+   */
+  isRunning(): boolean {
+    return this.viteDevServer !== null;
+  }
+
+  /**
+   * Dynamically loads Vite module
+   */
+  private async loadVite(): Promise<any> {
+    try {
+      // Vite is an ESM module, so we need to use dynamic import
+      // This requires Jest to be configured with ESM support
+      // @ts-expect-error - Vite is an optional peer dependency
+      const viteModule = await import('vite');
+      return viteModule;
+    } catch {
+      return null;
+    }
+  }
+
+  /**
+   * Creates Vite configuration optimized for Jest watch mode
+   */
+  /**
+   * Creates Vite configuration optimized for Jest watch mode
+   * Merges user-provided config with sensible defaults
+   */
+  private async createViteConfig(): Promise<any> {
+    const vite = await this.loadVite();
+    
+    const baseConfig: any = {
+      // Optimize for watch mode
+      optimizeDeps: {
+        disabled: false,
+      },
+      // Allow configuring resolve to match Jest's module resolution
+      resolve: {
+        conditions: ['node', 'default'],
+      },
+      root: this.projectRoot,
+      server: {
+        hmr: true, // Enable Hot Module Replacement
+        port: 5173,
+        strictPort: false, // Allow fallback to another port
+      },
+    };
+
+    // Merge user config from Jest config with base config
+    // User config takes precedence over base config
+    return vite.mergeConfig(baseConfig, this.config);
+  }
+}
+
+/**
+ * Helper to extract Vite watch mode config from Jest config
+ */
+/**
+ * Helper to extract Vite watch mode config from Jest config
+ * The experimental_vite object can contain any Vite configuration options
+ * that will be passed directly to Vite's createServer API
+ * Supports both ProjectConfig and direct experimental_vite config
+ */
+export function getViteWatchModeConfig(
```

Comment:
```suggestion
export function resolveViteConfig(
```

We would need to reuse this function for non-watch mode as well, don't you think?